### PR TITLE
Feature/registration

### PR DESCRIPTION
**Pull request by sahiga**

### CODE REVIEW

**Comment by sahiga** [comment]
@rbuckheit - can you review this PR when you get a chance? Thanks!

File: dist/kryptnostic.js [hunk 1]
```diff
@@ -24654,14 +24654,10 @@ define("revalidator", function(){});
```
Comment (rbuckheit) [inline dist/kryptnostic.js]: the situation with `dist` is a little funky - those files only get updated when we do a release of the product. once we merge your change we can do a release together so you know how to do that (the script for this is `bump.rb`). for now, can you revert the changes on the files in `dist` and just commit the source files?

File: js/src/http/RegistrationApi.coffee [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      Promise.resolve(axios({
+        method : 'POST'
+        data :
+          realm       : realm
+          name        : email
+          certificate : ''
+          email       : email
+          givenName   : givenName
+        url : registrationUrl()
+        }))
```
Comment (rbuckheit) [inline js/src/http/RegistrationApi.coffee]: it looks like the indentation is off on the closing `}))`

File: js/src/http/RegistrationApi.coffee [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  # HTTP calls for registration.
+  #
+  class RegistrationApi
+
+    register: (realm, email, givenName) ->
+      Promise.resolve(axios({
+        method : 'POST'
+        data :
+          realm       : realm
+          name        : email
```
Comment (rbuckheit) [inline js/src/http/RegistrationApi.coffee]: is using email here intentional or should there be a separate name parameter?
I tend to like defining all the variables up front and using the cs structuring/destructuring to create objects:
`data: { realm, name, email, givenName }`

thought that might help with the alignment here, but that's a matter of preference so do whatever makes it more readable to you

Comment (sahiga) [inline js/src/http/RegistrationApi.coffee]: I think we're still using email addresses as usernames. I copied the request from kryptnostic-auth.js:

```
     var request = {
        method: 'POST',
        data: {
          realm       : realm,
          name        : email,
          certificate : '',
          email       : email,
          givenName   : givenName
        },
        url : REGISTRATION_PATH
      };
```

File: js/src/service/RegistrationClient.coffee [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+  #
+  # Allows user to register for a new account.
+  #
+  class RegistrationClient
+
+    constructor : ->
+      @registrationApi = new RegistrationApi()
+
+    register : (realm, email, givenName) ->
```
Comment (rbuckheit) [inline js/src/service/RegistrationClient.coffee]: we've used a bunch of different terms for auth-related parameters, so I standardized everything in this repo. The terminology i've been using is:
`username` for the user's login (i think this corresponds to `email`)
`realm` for realm (good to go)
`password` for plaintext passwords
`credential` for the credential derived from password and encrypted salt
`principal` for string of the form `realm|username`
could you update these args to match?

I've also been using destructuring in order to require callers to name their arguments when using auth-related components (it makes refactoring easier when you add/remove new arguments from the calls and prevents bugs caused by ordering), would you be cool with changing this (and the function in `registrationApi`) to:
`register: ({ ream, username, name }) ->`?

these are all conventions I made up by myself, so if any of them don't make sense or you have ideas for better ways let me know, i'm all for it!

**Comment by rbuckheit** [comment]
finished my review - overall looks great just had a few comments on naming and things I wasn't sure about. LMK what you think and let me know if you push an update so I can take a look and then we can merge your change and build a new release.

File: js/src/http/RegistrationApi.coffee (hunk 3 above)
Comment (rbuckheit) [inline js/src/http/RegistrationApi.coffee]: sounds good, just wanted to make sure. thanks for checking!

**Comment by sahiga** [comment]
Thanks for the detailed review, @rbuckheit. I renamed the parameters and added destructuring. Let me know if you think of any other changes I should make.

File: package.json [hunk 5]
```diff
@@ -12,7 +12,8 @@
   },
   "contributors": [
     "Nick Hewitt <nick@kryptnostic.com>",
-    "Ryan Buckheit <ryan@kryptnostic.com>"
+    "Ryan Buckheit <ryan@kryptnostic.com>",
+    "Stephanie A. Higa <stephanie@kryptnostic.com"
```
Comment (rbuckheit) [inline package.json]: missing a closing `>` here

**Comment by rbuckheit** [comment]
LGTM! feel free to merge after fixing the `>`
